portfolio/src/main/webapp/script.js

```diff
@@ -45,26 +45,61 @@ function openConnectModal() {
   modal.style.display = "block";
 }
 
-function closeIntroModal(){
+function openQuestionModal() {
+  const modal = document.getElementById("myQuestionModal");
+  modal.style.display = "block";
+}
+
+function openViewModal() {
+  const modal = document.getElementById("myQuestionViewModal");
+  modal.style.display = "block";
+  fetch('/data').then(response => response.json()).then((Questions) => {
+  const questionContainer = document.getElementById("question-container");
+  questionContainer.innerHTML = '';
+  for (var i = 0; i < Questions.length; i++) {
```

Comment:
can you use `let` instead of `var` for i?  it's a variable that's being reassigned, and `var` is discouraged because of weird scoping behavior.